src/conduit/ConduitClient.php

```diff
@@ -163,6 +184,55 @@ public function callMethod($method, array $params) {
     return $conduit_future;
   }
 
+  // UBER CODE BEGIN
+  private function addWonkaClaimToCookies($cookies) {
+    $claim = $this->getWonkaClaim();
+    $new_cookies = '';
+    if (empty($claim)) {
+      return $cookies;
+    }
+    $usso_cookie = "usso={$claim}";
+    if (empty($cookies)) {
+      $new_cookies = $usso_cookie;
+    } else {
+      $new_cookies = $cookies."; {$usso_cookie}";
+    }
+    return $new_cookies;
+  }
+
+  private function getWonkaClaim() {
+    $claim = '';
+    // if the file exists and is current, use it.
+    $claim_file = getenv('ARC_USSO_COOKIE');
+    if (file_exists($claim_file)) {
+      $stale = time() - filemtime($claim_file);
+      if ($stale < 60) {
+        $claim = $this->readClaimFile($claim_file);
+      }
+    }
+    if (empty($claim)) {
+      $wonka_cli = getenv('ARC_WONKA_CLI');
+      system("{$wonka_cli} {$claim_file}");
```

Comment:
Hey @Pawka,

Thanks for your review here. I'm curious if there are libphutil use cases that I'm not familiar with.

I'm familiar with the arc use case, in which arc is running within the same authentication context as the calling user. In that case, I don't believe this is a security vulnerability. Whatever command that is called via `system` could just as easily be called directly. Or be added to PROMPT variables, etc.

If libphutil is used in server side requests, or embedded in other applications that cross authentication contexts (suid, etc) I would agree that this should be revisited.

How would you approach the desire to not hardcode the wonka cli? Is there a config file it can be added to? (Is where that config file is read from stored in an env?)